### PR TITLE
Add logging in namespace replication error

### PR DESCRIPTION
**Pull request by yux0**
## What changed?
1. Update logging in namespace replication error

## Why?
Debugging

### CODE REVIEW

File: common/namespace/replication_task_executor.go [hunk 1]
```diff
@@ -130,8 +130,11 @@ func (h *namespaceReplicationTaskExecutorImpl) shouldProcessTask(ctx context.Con
 	})
 	switch err.(type) {
 	case nil:
-		if resp.Namespace.Info.Id != task.GetId() {
-			h.logger.Error("namespace replication encountered name / UUID collision during determination on whether to process task", tag.WorkflowNamespaceID(task.Info.GetName()))
+		if resp.Namespace.Info.Id != task.GetInfo().GetId() {
```
Comment (yux0) [inline common/namespace/replication_task_executor.go:133]: This is the issue

Comment (yux0) [reply common/namespace/replication_task_executor.go]: The task.info.id is the same as task.id

File: common/namespace/replication_task_executor.go [hunk 2]
```diff
@@ -130,8 +130,10 @@ func (h *namespaceReplicationTaskExecutorImpl) shouldProcessTask(ctx context.Con
 	})
 	switch err.(type) {
 	case nil:
-		if resp.Namespace.Info.Id != task.GetId() {
-			h.logger.Error("namespace replication encountered name / UUID collision during determination on whether to process task", tag.WorkflowNamespaceID(task.Info.GetName()))
+		if resp.Namespace.Info.GetId() != task.GetId() {
+			h.logger.Error(
+				"namespace replication encountered UUID collision processing NamespaceCreationReplicationTask",
+				tag.WorkflowNamespaceIDs(map[string]struct{}{resp.Namespace.Info.Id: {}, task.GetId(): {}, task.Info.GetId(): {}}))
```
Comment (hehaifengcn) [inline common/namespace/replication_task_executor.go:136]: can we print `resp.Namespace.Info.GetId() and task.GetId()` explicitly in the error message? same applies to change at line 226 below.

File: common/namespace/replication_task_executor.go [hunk 3]
```diff
@@ -130,8 +130,10 @@ func (h *namespaceReplicationTaskExecutorImpl) shouldProcessTask(ctx context.Con
 	})
 	switch err.(type) {
 	case nil:
-		if resp.Namespace.Info.Id != task.GetId() {
-			h.logger.Error("namespace replication encountered name / UUID collision during determination on whether to process task", tag.WorkflowNamespaceID(task.Info.GetName()))
+		if resp.Namespace.Info.GetId() != task.GetId() {
```
Comment (hehaifengcn) [inline common/namespace/replication_task_executor.go:133]: should we change to use `resp.Namespace.Info.GetId()` in other place as well?

File: common/namespace/replication_task_executor.go [hunk 4]
```diff
@@ -214,7 +222,10 @@ func (h *namespaceReplicationTaskExecutorImpl) handleNamespaceCreationReplicatio
 		switch getErr.(type) {
 		case nil:
 			if resp.Namespace.Info.Name != task.Info.GetName() {
-				h.logger.Error("namespace replication encountered name / UUID collision during NamespaceCreationReplicationTask (second GetNamespace call)", tag.WorkflowNamespaceID(task.Info.GetName()), tag.NewErrorTag(err), tag.NewErrorTag(getErr))
+				h.logger.Error(
```
Comment (hehaifengcn) [inline common/namespace/replication_task_executor.go:225]: why remove `second GetNamespace call)`?

File: common/namespace/replication_task_executor.go (hunk 3 above)
Comment (yux0) [reply common/namespace/replication_task_executor.go:133]: same thing. I am reverting this one.

File: common/namespace/replication_task_executor.go (hunk 4 above)
Comment (yux0) [reply common/namespace/replication_task_executor.go:225]: I think it is redundant.

**Review by yycptt — APPROVED** [review]

File: common/namespace/replication_task_executor.go [hunk 5]
```diff
@@ -131,7 +131,11 @@ func (h *namespaceReplicationTaskExecutorImpl) shouldProcessTask(ctx context.Con
 	switch err.(type) {
 	case nil:
 		if resp.Namespace.Info.Id != task.GetId() {
-			h.logger.Error("namespace replication encountered name / UUID collision during determination on whether to process task", tag.WorkflowNamespaceID(task.Info.GetName()))
+			h.logger.Error(
+				"namespace replication encountered UUID collision processing namespace replication task",
+				tag.WorkflowNamespaceID(resp.Namespace.Info.Id),
+				tag.NewStringTag("Task Namespace Id", task.GetId()),
+				tag.NewStringTag("Task Namepsace Info Id", task.Info.GetId()))
```
Comment (kevinkyyro) [inline common/namespace/replication_task_executor.go:138]: Noticed this typo: `s/Namepsace/Namespace/`

**Pull request merged**